assets/js/googlesitekit/datastore/user/authentication.js

```diff
@@ -0,0 +1,233 @@
+	switch ( type ) {
+		case START_FETCH_AUTHENTICATION: {
+			return {
+				...state,
+				isFetchingAuthentication: true,
+			};
+		}
+
+		case RECEIVE_AUTHENTICATION: {
+			const { authentication } = payload;
+
+			return {
+				...state,
+				authentication,
+			};
+		}
+
+		case FINISH_FETCH_AUTHENTICATION: {
+			return {
+				...state,
+				isFetchingAuthentication: false,
+			};
+		}
+
+		case CATCH_FETCH_AUTHENTICATION: {
+			return {
+				...state,
+				error: payload.error,
+				isFetchingAuthentication: false,
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+export const selectors = {
+
+	/**
+	 * Gets the authentication info for this user.
+	 *
+	 * Returns `undefined` if the authentication info is not available/loaded.
+	 *
+	 * Returns an object with the shape when successful:
+	 * ```
+	 * {
+	 *   authenticated: <Boolean>,
+	 *   grantedScopes: <Array>,
+	 *   requiredScopes: <Array>
+	 * }
+	 * ```
+	 *
+	 * @private
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Object|undefined)} User authentication info.
+	 */
+	getAuthentication( state ) {
+		const { authentication } = state;
+		return authentication;
+	},
+	/**
+	 * Gets the Site Kit authentication status for this user.
+	 *
+	 * Returns `true` if the user is authenticated, `false` if
+	 * not. Returns `undefined` if the authentication info is not available/loaded.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(boolean|undefined)} Site connection status.
+	 */
+	isAuthenticated: createRegistrySelector( ( select ) => () => {
+		const { authenticated } = select( STORE_NAME ).getAuthentication() || {};
+		return authenticated;
+	} ),
+
+	/**
+	 * Gets the granted scopes for the user.
+	 *
+	 * Returns an array of granted scopes or undefined
+	 * if authentication info is not available/loaded.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Array|undefined)} Array of granted scopes
+	 */
+	getGrantedScopes: createRegistrySelector( ( select ) => () => {
+		const { grantedScopes } = select( STORE_NAME ).getAuthentication() || {};
+		return grantedScopes;
+	} ),
+
+	/**
+	 * Gets the required scopes for the user.
+	 *
+	 * Returns an array of required scopes or undefined
+	 * if authentication info is not available/loaded.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Array|undefined)} Array of granted scopes
+	 */
+	getRequiredScopes: createRegistrySelector( ( select ) => () => {
+		const { requiredScopes } = select( STORE_NAME ).getAuthentication() || {};
+		return requiredScopes;
+	} ),
+};
+
+export const resolvers = {
+	*getAuthentication() {
+		const registry = yield Data.commonActions.getRegistry();
+
+		const authenticationInfo = registry.select( STORE_NAME ).getAuthentication();
+
+		if ( ! authenticationInfo ) {
+			yield actions.fetchAuthentication();
+		}
+	},
+};
```

Comment:
Nitpicking, but can we put this `resolvers` block above `selectors` like elsewhere?